src/AssimpLoader.cpp

```diff
@@ -1390,6 +1392,13 @@ bool AssimpLoader::createSubMesh(const Ogre::String& name, int index, const aiNo
 
             for (size_t i=0; i < mesh->mNumFaces;++i)
             {
+                    if(faces->mNumIndices != 3) {
```

Comment:
this means that we got points or lines here instead of triangles as we expect. See:
http://assimp.sourceforge.net/lib_html/mesh_8h.html#ac352902280db7c3bd8ba64c133b7f03f

your workaround simply discards them. While we can merge this as is for now, please add a comment why the check is there and what it is doing.